modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/rel/IndexScanNode.java

```diff
@@ -117,19 +124,21 @@ protected Publisher<RowT> scan() {
     }
 
     private Publisher<RowT> indexPublisher(int[] parts, @Nullable RangeCondition<RowT> cond) {
-        Iterator<Publisher<? extends RowT>> it = new TransformingIterator<>(
-                Arrays.stream(parts).iterator(),
-                part -> partitionPublisher(part, cond)
-        );
+        boolean roTx = context().transactionTime() != null;
+        List<Publisher<? extends RowT>> publishers = new ArrayList<>(parts.length);
+
+        for (int i = 0; i < parts.length; i++) {
+            publishers.add(partitionPublisher(parts[i], roTx ? -1 : terms[i], cond));
```

Comment:
the change removed the lazy opening of the cursors

Reply:
Just to use `IntStream.range(0, parts.length).iterator()` as source for array indexes

Reply:
Fixed, thanks.